tools/src/launcher/Main.hs

```diff
@@ -128,12 +129,20 @@ data NodeData = NodeData
     , ndArgs    :: ![Text]
     , ndLogPath :: Maybe FilePath }
 
--- | Updater path, args, windows runner path, archive path
-data UpdaterData = UpdaterData
-    { udPath        :: !FilePath
-    , udArgs        :: ![Text]
-    , udWindowsPath :: Maybe FilePath
-    , udArchivePath :: Maybe FilePath
+data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OSXUD | UnsupportedOSUpdateData UnsupOSUD
```

Comment:
I think `UnsupOSUD` is useless. You can write `data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OSXUD | UnsupportedOSUpdateData`.